packages/eslint-plugin-react-hooks/__tests__/ESLintRuleExhaustiveDeps-test.js

```diff
@@ -2224,7 +2224,189 @@ const tests = {
         }
       `,
       errors: [
-        // TODO: make this message clearer since it's not obvious why.
+        "React Hook useEffect has a missing dependency: 'props'. " +
```

Comment:
This error message kinda surprises me when taking into account our recent discussion ( https://github.com/facebook/react/pull/14876 ) around props referential equality etc. 

If `props` have "unstable" reference then IMHO you shouldnt ever suggest using it as hook's dependency 🤔 The latter advice sounds way more appropriate (destructuring)

Reply:
`props` is OK as an escape hatch when you just want the rule to shut up. It's still stable if you only set state, for example. Better than nothing.

Reply:
Shouldnt the advice get reordered then? First propose destructuring and only latter propose using `props` directly.

You are calling it an escape hatch to shut up the rule - so it sounds to me that this is actually not something which you'd like to recommend.

Reply:
You're welcome to send a PR. :-) There are more worrying things I'm currently focused on than this particular message.

Reply:
Sure, might prepare one - was just validating my thinking so far :)